Sources/SkipUI/SkipUI/Text/TextField.swift

```diff
@@ -107,25 +111,39 @@ public struct TextField : View, Renderable {
         let colors = Self.colors(styleInfo: styleInfo)
         let keyboardOptions = isSecure ? KeyboardOptions(keyboardType = KeyboardType.Password) : EnvironmentValues.shared._keyboardOptions ?? KeyboardOptions.Default
         let keyboardActions = KeyboardActions(EnvironmentValues.shared._onSubmitState, LocalFocusManager.current)
-
+        
         let visualTransformation = isSecure ? PasswordVisualTransformation() : VisualTransformation.None
+        
         let currentText = text.wrappedValue
-        let defaultTextFieldValue = TextFieldValue(text: currentText, selection: TextRange(currentText.count))
+        let currentSelection = selection?.wrappedValue?.asComposeTextRange()
+        let defaultTextFieldValue = TextFieldValue(text: currentText)
         let textFieldValue = remember { mutableStateOf(defaultTextFieldValue) }
-        var currentTextFieldValue = textFieldValue.value
-        // If the text has been updated externally, use the default value for the current text,
-        // which also places the cursor at the end. This mimics SwiftUI behavior for external modifications,
-        // such as when applying formatting to the user input
-        if currentTextFieldValue.text != currentText {
-            currentTextFieldValue = defaultTextFieldValue
+        
+        LaunchedEffect(currentText) {
+            var value = textFieldValue.value
+            if value.text != currentText {
+                let selection = currentSelection ?? TextRange(currentText.count)
+                textFieldValue.value = TextFieldValue(text = currentText, selection = selection)
+            }
         }
-
+        
+        LaunchedEffect(currentSelection) {
```

Comment:
Same question as above, and also why does this need to be a _separate_ LaunchedEffect than the previous one?